Dockerfile

```diff
@@ -1,6 +1,28 @@
+# Use a UUID as placeholder value to have a unique string to replace. 
+ARG BASE_URL_PLACEHOLDER=189b303e-37a0-4f6f-8c0a-50333bc3c36e
+
 FROM docker.io/library/node:16.13.2 as build
+
+ARG BASE_URL_PLACEHOLDER
+
+COPY package.json package-lock.json ./
+RUN npm install --frozen-lockfile
+
 COPY ./ ./
-RUN npm install && npm run build
+# Set the React PUBLIC_URL to our placeholder value so that
+# that it can easily be replaced with the actual base URL
+# in the entrypoint script below.
+RUN PUBLIC_URL=${BASE_URL_PLACEHOLDER} npm run build
+
+FROM docker.io/library/nginx:1.21.5-alpine as runtime
+
+ARG BASE_URL_PLACEHOLDER
+ARG ENTRYPOINT_SCRIPT=/docker-entrypoint.d/set-public-url.sh
```

Comment:
Could we add info about `docker-entrypoint.d`? Tbh I had to read about it, because I didn't know about that 😅  You know, for future contributors to help them understand that step.

Reply:
Yes, sure! I added a comment!